### PR TITLE
EMCB-102: Support Bitbucket Server

### PR DESCRIPTION
**Pull request by ragrus-nbl**
Testing results before the changes: `535 specs, 166 failures, 8 pending specs`
Testing results after the changes: `535 specs, 163 failures, 8 pending specs`

But the changes shouldn't affect the existing tests. So it looks like some tests fail randomly.

Manual testing was only done against the public EI cloud under Windows. Eaton's private Bitbucket Server was used during the testing.

Additional notes:
- `impt test update` command doesn't work with the following message "This command does NOT work currently"
- If there is no field "builder" in the project file, `impt build run` command fails with an error like "Cannot read property 'libs' of undefined". And by default, `impt project` commands don't add this field when creating the project file.
- `impt test delete -i` command doesn't remove the GitHub credentials file referenced by test configuration file. This concerns the `impt test delete -b` (for the Bitbucket Server credentials file) command too.

### CODE REVIEW

File: CommandsManual.md [hunk 1]
```diff
@@ -1794,9 +1861,9 @@ Updates the specified webhook with a new target URL and/or MIME content-type. Fa
 
 | Command<br />Option<br />Alias | Command Option<br />Full Name(s) |
 | --- | --- |
-| -a | --all, --assigned, --allow-disconnect |
+| -a | --all, --assigned, --allow-disconnect, --bitbucket-srv-addr |
 | -ac | --account |
-| -b | --build, --builds |
+| -b | --build, --builds, --bitbucket-srv-config |
```
Comment (mm-gmbd) [inline CommandsManual.md:1866]: I'm confused by this one... (and the `-e` flag). If I specify the `-b` flag, which command does it actually infer, `--build`, `--builds`, or `--bitbucket-srv-config`.

File: package.json [hunk 2]
```diff
@@ -34,7 +34,7 @@
   },
   "homepage": "https://github.com/electricimp/imp-central-impt",
   "dependencies": {
-    "Builder": "https://github.com/deldrid1/Builder#hotfix/remote-relative-includes",
+    "Builder": "https://github.com/EatonGMBD/Builder#develop",
```
Comment (mm-gmbd) [inline package.json:37]: Should this be a dependency on `develop` or should this be a versioned released (even if it's marked as a pre-release?)

**Review by mm-gmbd — COMMENTED** [review]
A couple of comments, one question specifically about the version of Builder that's included in `package.json`?

File: CommandsManual.md (hunk 1 above)
Comment (ragrus-nbl) [reply CommandsManual.md:1866]: It depends on the command you are invoking. For example, if you write `impt build copy -b xxx`, then it means `--build`. But if you write `impt test bitbucket-server -b xxx`, then it means `--bitbucket-srv-config`.
Aliases are made the way they don't interfere with each other. I.e., there are no commands where both `--build` and `--bitbucket-srv-config` can be used.

File: package.json (hunk 2 above)
Comment (ragrus-nbl) [reply package.json:37]: It is just a temporary dependency. Once Builder is released, we will change the dependency here to a tagged released version.

**Comment by mm-gmbd** [comment]
Approved!

**Pull request merged**